runtime/interpreter/value.go

```diff
@@ -11541,14 +11541,11 @@ func NewPublicKeyValue(
 		sema.PublicKeyVerifyPoPFunction: publicKeyVerifyPoPFunction,
 	}
 
-	// Validate the public key, and initialize 'isValid' field.
-
-	publicKeyValue.SetMember(
-		interpreter,
-		getLocationRange,
-		sema.PublicKeyIsValidField,
-		validatePublicKey(interpreter, getLocationRange, publicKeyValue),
-	)
+	// Validate the public key.
+	isValid := validatePublicKey(interpreter, getLocationRange, publicKeyValue)
+	if !isValid {
+		panic("invalid public key")
```

Comment:
Instead of using a string here, declare a type in `runtime/errors.go` and make it conform to `error`. In addition, provide position information. The easiest way to do that is by embedding `interpreter.LocationRange` and passing it from here by calling `getLocationRange()`

Reply:
What should I store in that struct?

I tried to settle on using a string because the places in which I need to instantiate `InvalidPublicKey` do not have the same data type for PublicKey. It's either `ArrayValue` or `CompositeValue`, though `[]byte` seems ideal. There's code to convert `ArrayValue` to `CompositeValue` for `PublicKey` but it does validation so I would need to refactor it.

But using a string causes the linter to fail because I use `fmt.Sprintf("%s", foo)`. Unfortunately there is no `String(foo)` like it suggests and `string(foo)` can't actually serialize `ArrayValue`.

I made a hack around this by adding a space like `%s ` but that's not great.

I suppose I could instead make the error struct take several types and have it print the first one that isn't `nil`... thoughts?

---

linter error:
https://github.com/onflow/cadence/runs/4388776276?check_suite_focus=true#step:5:24